# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorEGL.cpp
# Commit: c766791726f1
# Full Hash: c766791726f1bbce2bb7581c62ccc8b334e8c9e2
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-06 04:08:03
# Description:
#   Bug 1943662 [Wayland] Don't call GdkCeiledScaleFactor() from compositor/render threads r=emilio
#   
#   - Mark SetEGLNativeWindowSize() calls Wayland only as we need to set explicitly wl_egl_window size on Wayland.
#   - Remove moz_container_wayland_egl_window_set_size() and use WaylandSurface directly.
#   - Remove plain call of FractionalScaleFactor() from nsWindow::SetEGLNativeWindowSize() as it used for logging only and move it to logging block.
# ==============================================================================

diff -r 12d7b2544b10 -r c766791726f1 gfx/webrender_bindings/RenderCompositorEGL.cpp
--- a/gfx/webrender_bindings/RenderCompositorEGL.cpp	Wed Feb 05 16:34:23 2025 +0000
+++ b/gfx/webrender_bindings/RenderCompositorEGL.cpp	Wed Feb 05 16:38:14 2025 +0000
@@ -88,11 +88,11 @@
         << "We don't have EGLSurface to draw into. Called too early?";
     return false;
   }
-#ifdef MOZ_WIDGET_GTK
+#ifdef MOZ_WAYLAND
   if (mWidget->AsGTK()) {
     if (!mWidget->AsGTK()->SetEGLNativeWindowSize(GetBufferSize())) {
-      // It's possible that GtkWidget is hidden on Wayland; e.g. maybe it's
-      // just been closed. So, we can't draw into it right now.
+      // Wayland only check we have correct window size to avoid
+      // rendering artifacts.
       return false;
     }
   }